File: java/codegym/mvc_architecture/src/model/Product.java
package model;

public class Product {
    private int id;
    private String name;

    public Product(String name) {
        this.name = name;
    }

    public int getId() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        return id == ((Product)o).id ;
    }

    @Override
    public String toString() {
        return "Product{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: pattern/csharp/demo_delegate/demo_delegate/model/Product.cs
using System;
namespace demo_delegate.model
{
    public class Product
    {
        private int id;
        private String name;
        private String color;
        private double weight;
        private double price;

        public Product()
        {
        }

        public Product(int id, string name, string color, double weight, double price)
        {
            this.id = id;
            this.name = name;
            this.color = color;
            this.weight = weight;
            this.price = price;
        }

        public void setId(int id)
        {
            this.id = id;
        }

        public int getId()
        {
            return id;
        }

        public double getPrice()
        {
            return price;
        }

        public double getWeight()
        {
            return weight;
        }

        
    public override String ToString()
        {
            return "Product{" +
                    "id=" + id +
                    ", name='" + name + '\'' +
                    ", color='" + color + '\'' +
                    ", weight=" + weight +
                    ", price=" + price +
                    '}';
        }
    }
}
